make/compiler/Makefile.gnu

```diff
@@ -90,12 +90,24 @@ endif
 #
 # query gcc version
 #
+ifndef GNU_GCC_MAJOR_VERSION
+export GNU_GCC_MAJOR_VERSION = $(shell $(CC) -dumpversion | awk '{split($$1,a,"."); printf("%s", a[1]);}')
```

Comment:
Using cut should work fine too.  I just wanted my changes to use the identical mechanism to what the Makefile already used, so I would introduce a minimum of possibilities for errors.